File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/analysis/acceptance-project/src/surveycrawler/guarded_run.py
from surveycrawler.crawler_run import CrawlerRun


class GuardedRun(CrawlerRun):

    # initialization

    def __init__(self):
        self._movements = []

    # command processing

    def start_guarded_run(self, a_crawler):
        a_crawler.signal_guarded_run_already_started()

    def finish_guarded_run(self, a_crawler):
        a_crawler.finish_guarded_run()

    def assert_is_finished(self, a_crawler):
        a_crawler.signal_unfinished_guarded_run()

    # moving

    def slide_crawler_towards(self, a_crawler, a_direction, a_silt):
        a_crawler.signal_can_not_slide_during_guarded_run()

    # undoing

    def register_movement(self, a_movement):
        self._movements.append(a_movement)

    def undo_movements(self, a_crawler):
        while self._movements:
            self._movements.pop().undo(a_crawler)

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-212555-19812/output/project/src/main/java/surveycrawler/GuardedRun.java
package surveycrawler;

import java.util.ArrayList;
import java.util.List;

public class GuardedRun extends CrawlerRun {

    private final List<CrawlerMovement> movements = new ArrayList<>();

    // movements

    @Override
    public void registerMovement(CrawlerMovement aMovement) {
        movements.add(aMovement);
    }

    @Override
    public void moveUnpredictably(SurveyCrawler aCrawler, Point aDirection, int aNumberOfCells) {
        aCrawler.signalUnpredictableMovement();
    }

    @Override
    public void undoMovementsOn(SurveyCrawler aCrawler) {
        for (int i = movements.size() - 1; i >= 0; i--) {
            movements.get(i).undoOn(aCrawler);
        }
        movements.clear();
    }

    // guarded runs

    @Override
    public void startGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.signalGuardedRunAlreadyStarted();
    }

    @Override
    public void endGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.finishGuardedRun();
    }

    @Override
    public void assertIsFinished(SurveyCrawler aCrawler) {
        aCrawler.signalGuardedRunNotFinished();
    }
}
